src/pytorch_lightning/core/module.py

```diff
@@ -1585,6 +1585,10 @@ def lr_scheduler_step(self, scheduler, optimizer_idx, metric):
                 scheduler.step(epoch=self.current_epoch)
 
         """
+        optimizer = self.trainer.optimizers[optimizer_idx]
+        if hasattr(optimizer, "_step_count") and optimizer._step_count <= 0:
```

Comment:
This only works for the first step but the warning can appear after any optimizer step, as there's no easy way to know whether the `optimizer.step` was skipped by AMP.

Reply:
This works for the N first "steps" that are skipped. True that we can't detect skipped steps passed the first true step, but PyT doesn't print any warning for those https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L138

Reply:
> True that we can't easily detect skipped steps passed the first true step, but PyT doesn't print any warning for those

No, but you can detect it by comparing the scale factor before and after the `scaler.step()` as I wrote in the linked issue. IMHO, I wouldn't see the current change as a solution for the reason @carmocca stated.

Reply:
Ok, I can change the implementation to a comparison of scales if that's prefered

Reply:
Done